addon/models/resource.js

```diff
@@ -2,16 +2,45 @@ import Ember from 'ember';
 import { pluralize } from 'ember-inflector';
 
 const Resource = Ember.Object.extend({
+
+  /*
+    Defaults for a JSON API Resource object
+  */
+  id: null,
   type: null,
-  links: {},
-  attributes: {},
-  relationships: {},
+  links: null,
+  attributes: null,
+  relationships: null,
+
+  /*
+    init a JSON API Resource object with default as empty objects on the instance
+  */
+  init() {
+    const attrs = this.getProperties('links', 'attributes', 'relationships');
+    this.setProperties({
+      'links': attrs.links || {},
+      'attributes': attrs.attributes || {},
```

Comment:
@barelyknown attrs.attributes would be what was passed into create. The problem was that `Ember.Object.extend({ attributes: {} })` when injected in the container somehow was sticky even with `{ instantiate: false, singleton: false }`. I recall @stefanpenner showing at Ember Conf that in the future we'll go back to the more idomatic way of using Constructors.